### PR TITLE
[WIP] rpms: separate specs by build target, clean dependencies

### PR DESCRIPTION
**Pull request by tpepper**
The one giant specfile is a mess and makes it unclear what are the
actual package dependencies.  This PR intends to splits them and makes a bunch
of cleanups.

[WIP] need to test again after splitting kubernetes-cni, and split into a more reviewable patch series.  the diffstat looks worse that it is due to copying one file into multiple.  Also compare / reconcile with what is in k/k/build today which is slightly more than here in k/release.  Then reconcile with the debs side of the house.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

### CODE REVIEW

**Comment by tpepper** [comment]
This is an alternative approach to https://github.com/kubernetes/release/pull/688

**Comment by tpepper** [comment]
/cc @spiffxp @timothysc @neolit123 @frapposelli @figo 
/cc @sumitranr @foxish @feiskyer @aleksandra-malinowska 

If we merge this, we need to build and release 1.11.10, 1.12.8, 1.13.6, 1.14.2.  And communicate to the userbase that the community supports on a best effort basis the current 1.X.Y.

**Comment by tpepper** [comment]
Also worth noting, I've not attempted to address the deb's in this PR.

**Comment by timothysc** [comment]
I'm a firm hold on this to put the specs in the main repo.

**Comment by tpepper** [comment]
File location aside,  @timothysc any comment on file content?

**Comment by spiffxp** [comment]
I'm a net "meh"

In terms of maintenance burden, I feel like splitting everything up makes it more difficult to understand what the differences between these are, and why.  It may lead to cargo-culting or copy-paste errors.  I don't know the tooling enough to say whether a different templating language might make things clearer, or would be a needless abstraction.

That said it's not clear to me why we need to block now, if this is decreasing pain.

File: rpm/kubelet_1.11.spec [hunk 1]
```diff
@@ -90,9 +69,9 @@ Command-line utility for interacting with a Kubernetes cluster.
 Version: %{KUBE_VERSION}
 Release: %{RPM_RELEASE}
 Summary: Command-line utility for administering a Kubernetes cluster.
-Requires: kubelet >= 1.6.0
-Requires: kubectl >= 1.6.0
-Requires: kubernetes-cni >= 0.7.5
+Requires: kubelet >= 1.10.0
+Requires: kubectl >= 1.10.0
```
Comment (spiffxp) [inline rpm/kubelet_1.11.spec]: Good sign of how often these are tended to

**Comment by tpepper** [comment]
TODO: reconcile with https://github.com/kubernetes/release/pull/734 on the deb side of the house

**Comment by spiffxp** [comment]
/uncc
let me me know when this is no longer wip

**Comment by tpepper** [comment]
For anybody interested, I'll keep my branch around where I'd been dabbling at this.  But this PR isn't going to merge.  There are multiple other people doing multiple other parallel hackings on the packaging scripts and all of our changes conflict architecturally if not also at a code line level.

**Pull request closed without merging**